File: hackerrank/problem solving/algorithms/2 - implementation/46 - beautiful triplets/beautiful_triplets_procedural.py
# Source: https://www.hackerrank.com/challenges/beautiful-triplets/problem?isFullScreen=true

def main():

    SIZE, BEAUTIFUL_DIFFERENCE = read_an_array()
    ARRAY = read_an_array()

    N_BEAUTIFUL_TRIPLETS = find_beautiful_triplets(ARRAY, BEAUTIFUL_DIFFERENCE)
    print(N_BEAUTIFUL_TRIPLETS)


def read_an_array():

    ARRAY = list(map(int, input().split()))
    return ARRAY


def find_beautiful_triplets(array, beautiful_difference):

    n_beautiful_triplets = 0

    if len(array) > 2:
        for i in range(0, len(array)-2):
            for j in range(i+1, len(array)-1):
                FIRST_DIFFERENCE = array[j] - array[i]

                if FIRST_DIFFERENCE < beautiful_difference:
                    continue
                elif FIRST_DIFFERENCE > beautiful_difference:
                    break
                
                for k in range(j+1, len(array)):
                    SECOND_DIFFERENCE = array[k] - array[j]

                    if SECOND_DIFFERENCE == beautiful_difference:
                        n_beautiful_triplets += 1
                    elif SECOND_DIFFERENCE > beautiful_difference:
                        break

    return n_beautiful_triplets


if __name__ == "__main__":
    main()

File: hackerrank/problem solving/algorithms/2 - implementation/46 - beautiful triplets/beautiful_triplets_procedural.cs
// Source: https://www.hackerrank.com/challenges/beautiful-triplets/problem?isFullScreen=true

using System;

public class BeautifulTriplets
{
    public static void Main() {
        List<int> inputLine = _readAnArray();
        int size = inputLine.First();
        int beautifulDifference = inputLine.Last();
        List<int> array = _readAnArray();

        int nBeautifulTriplets = _findBeautifulTriplets(array, beautifulDifference);
        Console.WriteLine(nBeautifulTriplets);
    }

        private static List<int> _readAnArray() {
            List<int> array = Console.ReadLine().Split().Select(int.Parse).ToList();
            return array;
        }

        private static int _findBeautifulTriplets(List<int> array, int beautifulDifference) {
            int nBeautifulTriplets = 0;

            if (array.Count > 2)
                for (int i = 0; i < array.Count - 2; i++)
                    for (int j = i + 1; j < array.Count - 1; j++) {
                        int firstDifference = array[j] - array[i];

                        if (firstDifference < beautifulDifference)
                            continue;
                        else if (firstDifference > beautifulDifference)
                            break;

                        for (int k = j + 1; k < array.Count; k++) {
                            int secondDifference = array[k] - array[j];

                            if (secondDifference == beautifulDifference)
                                nBeautifulTriplets++;
                            else if (secondDifference > beautifulDifference)
                                break;
                        }
                    }

            return nBeautifulTriplets;
        }
}
